src/GitServer/HttpDuplex.ts

```diff
@@ -0,0 +1,178 @@
+import http from 'http'
+import zlib from 'zlib';
+import through from 'through';
+import { EncryptedFS } from 'encryptedfs';
+import GitSideBand from "./side-band/GitSideBand";
+import packObjects from './pack-objects/PackObjects';
+
+const headerRE = {
+  'receive-pack': '([0-9a-fA-F]+) ([0-9a-fA-F]+) refs\/(heads|tags)\/(.*?)( |00|\u0000)|^(0000)$', // eslint-disable-line
+  'upload-pack': '^\\S+ ([0-9a-fA-F]+)'
+};
+
+export class HttpDuplex {
```

Comment:
What is the purpose of this class?

Reply:
This is to Handle the response when the client asks for a particular pack, to handle `POST` requests in other words.

Reply:
I've been able to reduce complexity enough to embed the whole of git server in a single `request.on('data', (data) => {})` method within the GitServer itself. So the `HttpDuplex class is now redundant and has been removed.